### PR TITLE
Fix sphinx build warnings indicating where extlinks can be used 

### PR DESCRIPTION
**Pull request by jodygarnett**
Most of the warnings were download instructions for extensions, so I just went through all of them to make it consistent.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by aaime** [comment]
Noice, the docs build is back to green. However, the CN doc build is still red:
image

Not sure what to do about the latter... would it be an easy fix? Or should we turn it off for the time being, and ask for help on the contributors to the CN translation?

**Comment by jodygarnett** [comment]
Noticed that also, it the zhcn docs should now build also

**Comment by jodygarnett** [comment]
And we are go: 
image

**Comment by aaime** [comment]
Yessir... I was trying to figure out where this magic "wiki" or "website" things were coming from, at first thought they were well-known bits of RST/Sphinx, but turns out they are [variables declared in configuration](https://github.com/geoserver/geoserver/blob/main/doc/en/user/source/conf.py#L107) and there are more of them.

Maybe I'm not the only one that had no idea about them...

File: doc/en/developer/source/maven-guide/index.rst [hunk 1]
```diff
@@ -198,31 +198,4 @@ This can also be used when running the local jetty application server:
 
   mvn jetty:run -DconfigId=release -DconfigDirectory=../../../data/release
 
-You may also use an absolute path, if you have a custom data directory you would like to use.
```
Comment (jodygarnett) [inline doc/en/developer/source/maven-guide/index.rst:201]: Looks like I removed one too many lines

**Comment by aaime** [comment]
Seeing the same build failures in stable and maintenance, a backport is needed.

**Pull request merged**